bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -456,6 +457,7 @@ private void maybeRecycle() {
     private void recyclePendAddOpObject() {
         entryId = LedgerHandle.INVALID_ENTRY_ID;
         currentLedgerLength = -1;
+        ReferenceCountUtil.release(payload);
```

Comment:
I think payload should only be released if `toSend` is not computed. if `toSend` is computed, the ref cnt of payload is either passed on to `toSend` (v2) or released (v3). unless I missed something from your last comment.